migrate/repo_checks.py

```diff
@@ -54,6 +75,93 @@ def dry_run(self):
         raise NotImplementedError
 
 
+class RequireTeamPermission(Check):
+    """
+    A generic check that is not meant to be used directly
+    but is used to reduce some copy-pasta.
+
+    Do not use directly, instead, subclass this class with
+    a specific team and permission level.
+    """
+
+    def __init__(self, api, org, repo, team, permission):
```

Comment:
Consider documenting the types / possible values of these arguments, either via type annotations or just a docstring. I figured out that `permission` was a string, but it took me a bit to get there, and I still don't know what values it could accept.